### PR TITLE
Fix link colors #14084 (Previous PR #14129)

### PR DESCRIPTION
**Pull request by damiend97**
## References
Addresses issue https://github.com/jupyterlab/jupyterlab/issues/14084
Reopens PR #14129 (accidental repo delete)

## Code changes
Changes to button and link text styles to make them more digestible

## User-facing changes
All of the changes are the same as listed in #14129
However - I felt that using the theme color variable on the about me buttons was more ideal than using white for both light and dark theme.

newfix

## Backwards-incompatible changes
None.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
cc @HaudinFlorence

**Review by andrii-i — COMMENTED** [review]
Thank you @damiend97. This fixes links + buttons looking funny in “About JupyterLab” popup (point 2 in #14084). I think this change makes them look much cleaner. 

Note that links in this popup are styled differently than links everywhere else in JupyterLab (other links don’t have a background on mouse hover, are colored blue rather than orange), we might want to address this and make them all look similar at some point (not necessarily in this PR).

Also note that point 1 in #14084 (links in warning dialogs like “Trust this notebook?” being unreadable on mouse hover in dark theme) still stands. @damiend97 do you plan to address it? It’s fine if you don’t, let’s just take a note  and make sure #14084 is not closed until point 1 is solved or a separate issue tracking it is created (CC @krassowski).

Screenshot 2023-03-10 at 2 27 32 PM

**Comment by JasonWeill** [comment]
I prefer the orange color of links in dark mode, although I think the black-on-red color scheme on hover is not necessary. Maybe, instead, we could slightly lighten or darken the link on hover. I'm also OK with not changing the color on hover.

**Comment by damiend97** [comment]
I think the biggest thing here is consistency. I agree with @JasonWeill that orange on black looks good. However, as @andrii-i mentioned, other links are styled blue rather than orange. What if we did orange links in dark mode and a visible light blue in light mode? Additionally, we could remove the background colors and text color changes on hover and only change the cursor.

**Comment by JasonWeill** [comment]
I'm also OK with using a lighter blue with high contrast. Try using the [WebAIM Contrast Checker](https://webaim.org/resources/contrastchecker/) to validate that your selected color has contrast that meets WCAG guidelines.

**Comment by damiend97** [comment]
The light blue I was intending on using is var(jp-content-link-color) - same color as var(--jp-brand-color1) which is used in other links sitewide. From what I can tell this translates to var(--md-blue-700) which is defined with material design specifications. Material designs stylesheet declares md-blue-700 to be #1976d2 (https://gist.github.com/frankyonnetti/fe8098d0295fcfe5d31f0f68c6fac323). WebAIM is saying that #1976d2 (light blue) on #fff (light mode) passes 4 out of 5 checks. #1976d2 on #212121 (dark mode) ( var(--jp-layout-color1) > var(--md-grey-900) also derived from Material Specs ) only passes 2 out of 5. I'm sorry if I'm making this more complicated than it should be but taking all of the above into consideration, the link color variables (site wide) do not comply to WebAIM. Or, I am not landing at the correct hex values for said variables.

To simplify, if this blue on white isn't acceptable...

linkAfterLight

then the vars (jp-content-link-color) and (jp-brand-color) should be adjusted. That is given we are to comply 100% with WebAIM.

Furthermore, the orange on black that you said you like @JasonWeill (--jp-warn-color1 > var(--md-orange-700) > #f57c00) passes 4 out of 5 on WebAim.

Sorry again if this is too much. Hopefully I'm not making things more difficult than they should be.

**Comment by JasonWeill** [comment]
@damiend97 In the dark theme, `--jp-content-link-color` is set to `var(--md-blue-300)`, which in turn is `#64b5f6`. This has a [contrast ratio of 7.27:1](https://webaim.org/resources/contrastchecker/?fcolor=64B5F6&bcolor=212121), which passes all WCAG requirements.

**Comment by andrii-i** [comment]
Based on my testing with Chrome dev tools they both don't pass 1 test. Good catch @damiend97.

Blue-on-white test [here](https://webaim.org/resources/contrastchecker/?fcolor=1976D2&bcolor=FFFFFF), screen below:
Screenshot 2023-03-10 at 5 32 36 PM

Orange-on-black test [here](https://webaim.org/resources/contrastchecker/?fcolor=F37726&bcolor=212121), screen below:
Screenshot 2023-03-10 at 5 37 35 PM

**Comment by damiend97** [comment]
My latest commits ensure WebAIM compliance with jp-Button-flat elements (within about popup) and all elements using jp-content-link-color (including jp-dialogue-content within "Trust this notebook").

file:///home/durandamien1997/Pictures/Screenshots/Screenshot%20from%202023-03-11%2008-31-46.png
file:///home/durandamien1997/Pictures/Screenshots/Screenshot%20from%202023-03-11%2008-31-24.png
file:///home/durandamien1997/Pictures/Screenshots/Screenshot%20from%202023-03-11%2008-29-09.png
file:///home/durandamien1997/Pictures/Screenshots/Screenshot%20from%202023-03-11%2008-30-40.png

File: packages/help-extension/style/base.css [hunk 1]
```diff
@@ -60,8 +60,14 @@
...
 
-.jp-About-externalLinks .jp-Button-flat {
-  color: #f37726;
+[data-jp-theme-light='false'] .jp-About-externalLinks .jp-Button-flat,
+[data-jp-theme-light='false'] .jp-About-externalLinks .jp-Button-flat:hover {
+  color: var(--jp-warn-color1);
+}
+
+[data-jp-theme-light='true'] .jp-About-externalLinks .jp-Button-flat,
+[data-jp-theme-light='true'] .jp-About-externalLinks .jp-Button-flat:hover {
```
Comment (krassowski) [inline packages/help-extension/style/base.css:69]: Let's simplify selectors for performance

```suggestion
.jp-About-externalLinks > .jp-Button-flat,
.jp-About-externalLinks > .jp-Button-flat:hover {
```

File: packages/help-extension/style/base.css [hunk 2]
```diff
@@ -60,8 +60,14 @@
   padding-top: 12px;
 }
 
-.jp-About-externalLinks .jp-Button-flat {
-  color: #f37726;
+[data-jp-theme-light='false'] .jp-About-externalLinks .jp-Button-flat,
+[data-jp-theme-light='false'] .jp-About-externalLinks .jp-Button-flat:hover {
```
Comment (krassowski) [inline packages/help-extension/style/base.css:64]:
```suggestion
[data-jp-theme-light='false'] .jp-About-externalLinks > .jp-Button-flat,
[data-jp-theme-light='false'] .jp-About-externalLinks > .jp-Button-flat:hover {
```

**Review by krassowski — COMMENTED** [review]
Small suggestions to simplify selectors and increase specificity (related to https://github.com/jupyterlab/jupyterlab/issues/14167)

**Comment by GabrielaVives** [comment]
I agree that the best choice would be color consistency for all themes, and finding maybe a lighter tone of blue for the dark theme as @damiend97 suggested would be a good option. 

I think that the choice between blue or orange links on a dark theme are independent of them being WCAG AAA compliant, because we can always find a contrast in these colors that passes the tests.
It is a design decision that can be quite subjective;
- orange aligned with the Jupyter branding and looks good on a dark theme as @andrii-i said
- blue is commonly used for links in other software, it will not surprise anyone

Whatever the choice is, it's important that it applies to **all** links in the theme to ensure a consistent design system.

**Comment by damiend97** [comment]
@GabrielaVives thanks for your input. I think it's best if we use blue links in both light and dark theme. It will offer consistency and simplicity. I will make the changes and make sure all links are the same color (or as close to each other as possible while adhering to WCAG). Hopefully we can all agree on that.

**Comment by andrii-i** [comment]
Taking in account WCAG AAA the contrast 7:1 and JupyterLab's default themes, Blue 800 (Hex Code: #1565C0) seems like the best option:

| Link Color (Blue) | Hex Code | JupyterLab Light Theme Background | Contrast Index (Light) | JupyterLab Dark Theme Background | Contrast Index (Dark) |
|-------------------|----------|-----------------------------------|-----------------------|----------------------------------|-----------------------|
| Blue 700          | #1E88E5  | #F5F5F5                            | 6.89                  | #212121                          | 7.07                   |
| Blue 800          | #1565C0  | #F5F5F5                            | 9.06                  | #212121                          | 9.28                   |
| Blue 900          | #0D47A1  | #F5F5F5                            | 11.70                 | #212121                          | 12.00                  |

**Comment by JasonWeill** [comment]
Will this change fix the problem in light mode where links have a red background on hover? Looks very jarring and low-contrast:

image

**Comment by andrii-i** [comment]
@JasonWeill yes, current version removes background color in [packages/application/style/buttons.css](https://github.com/jupyterlab/jupyterlab/pull/14159/files#diff-989497dbcc560b6ad6c598f818a8d35da2ad8d231ce4686a6b645c0aeb89b008)

**Comment by andrii-i** [comment]
Fixes #14084

**Comment by damiend97** [comment]
@andrii-i where are you getting your contrast index from? Both Google Dev Tools and WebAIM are giving me different results.

**Comment by damiend97** [comment]
@JasonWeill yes I have removed background colors on those links.

**Comment by damiend97** [comment]
Current state reflects the following...

| Element | Theme | Link Color | Background Color | Contrast Index |
| --------------- | --------------- | --------------- | --------------- | --------------- |
| About Popup | Light | `#0D47A1` | `#FFFFFF` | 8.63 |
| About Popup | Dark | `#64B5F6` | `#212121` | 7.27 |
| Trust Popup | Light | `#0D47A1` | `#FFFFFF` | 8.63 |
| Trust Popup | Dark | `#64B5F6` | `#212121` | 7.27 |

**Comment by damiend97** [comment]
@JasonWeill The current changes resolve issues #14084 and #14166. Additionally, red background on hover .jp-Button-flat has been removed. Colors now maintain consistency and adhere to WCAG contrast requirements. All checks have passed. I would say this PR should be ready to merge.

**Review by JasonWeill — APPROVED** [review]

**Comment by JasonWeill** [comment]
@damiend97 Thank you very much for your contribution!

**Pull request merged**